packages/iov-ethereum/src/wsjsonrpcclient.ts

```diff
@@ -0,0 +1,30 @@
+import { JsonRpcRequest, JsonRpcResponse, parseJsonRpcResponse2 } from "@iov/jsonrpc";
+import { StreamingSocket } from "@iov/socket";
+
+export class WsJsonRpcClient {
+  private readonly socket: StreamingSocket;
+
+  public constructor(socket: StreamingSocket) {
+    this.socket = socket;
+  }
+
+  public async run(request: JsonRpcRequest): Promise<JsonRpcResponse> {
+    await this.socket.connected;
+
+    const promise = new Promise<JsonRpcResponse>((resolve, reject) => {
+      this.socket.events
+        .map(event => parseJsonRpcResponse2(JSON.parse(event.data)))
```

Comment:
Parsing exceptions in here should be ignored. I suggest converting exception to `null` in the map body and then use a null filter like this one: https://github.com/iov-one/ponferrada/blob/95ec319/packages/sanes-chrome-extension/src/extension/background/model/persona/index.ts#L12-L14